version: "3.9"

services:
  mariadb:
    container_name: $DB_NAME
    image: mariadb:10.6.4-focal
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: $ROOT_PASSWORD
      MYSQL_USER: client
      MYSQL_PASSWORD: clientPassword5!
    volumes:
      - '$DATA_DIR:/var/lib/mysql/'
    ports:
      - $DB_PORT:3306
    networks:
      - dev-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "192.168.1.136", "--port", $DB_PORT, "-p$ROOT_PASSWORD"]
      interval: 30s
      timeout: 10s
      retries: 5

  init_db:
    container_name: init_db
    environment:
      ROOT_USERNAME: $ROOT_USERNAME
      ROOT_PASSWORD: $ROOT_PASSWORD
      HOSTNAME: $HOSTNAME
      DB_PORT: $DB_PORT
      DEBUG_MODE: $DEBUG_MODE
      API_PORT: $API_PORT
    restart: "no"
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - ../db:/db:rw
    networks:
      - dev-network
    depends_on:
      mariadb:
        condition: service_healthy
    command: python3 db/load_db.py $FORCE
    stdin_open: true
    tty: true

  server:
    container_name: $API_NAME
    restart: always
    environment:
      ROOT_USERNAME: $ROOT_USERNAME
      ROOT_PASSWORD: $ROOT_PASSWORD
      HOSTNAME: $HOSTNAME
      DB_PORT: $DB_PORT
      DEBUG_MODE: $DEBUG_MODE
      API_PORT: $API_PORT
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - ../server:/server:rw
    ports:
      - $API_PORT:5000
    networks:
      - dev-network
    depends_on:
      mariadb:
        condition: service_healthy
    command: python3 server/app.py
    stdin_open: true
    tty: true

networks:
  dev-network:
    driver: bridge